cpp/src/parquet/util/test-common.h

```diff
@@ -49,6 +50,28 @@ const char* get_data_dir() {
   return result;
 }
 
+std::string get_bad_data_dir() {
+  // PARQUET_TEST_DATA should point to ARROW_HOME/cpp/submodules/parquet-testing/data
+  // so need to reach one folder up to access the "bad_data" folder.
+  std::string data_dir(get_data_dir());
+  std::stringstream ss;
+  ss << data_dir << "/../bad_data";
+  return ss.str();
+}
+
+std::string get_data_file(const std::string& testing_file_path, bool is_good = true) {
+  std::stringstream ss;
+
+  if (is_good) {
+    ss << get_data_dir();
+  } else {
+    ss << get_bad_data_dir();
+  }
+
+  ss << "/" << testing_file_path;
```

Comment:
I'm new to this part of the code base, so I am not sure if this is really relevant, but are we expecting these tests to run on Windows? If so, will the use of forward slashes be an issue here when resolving the paths? Also, do we ever anticipate needing to support Unicode path names?

Reply:
I wasn't sure about this either and didn't have a windows machine handy to test this out on locally first.  Fortunately, AppVeyor runs these tests on windows and they passed without any platform-specific path massaging :smile:

Reply:
Thanks for that clarification!

Reply:
That said, I looked into using `boost::filesystem::path` to manage the path manipulation but couldn't work out how to easily link the test targets with the boost filesystem libs.

Reply:
Since this is minor, breaking off the use of `boost::filesystem::path` into a separate JIRA issue would probably make more sense. This doesn't seem too important since you mentioned that AppVeyor ran the tests without any issues.